nengo/spa/state.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+
+import nengo
+from nengo.spa.module import Module
+from nengo.networks.ensemblearray import EnsembleArray
+
+
+class State(Module):
+    """A module capable of representing a single vector, with optional memory.
+
+    This is a minimal SPA module, useful for passing data along (for example,
+    visual input).
+
+    Parameters
+    ----------
+    dimensions : int
+        Number of dimensions for the vector
+    subdimensions : int
+        Size of the individual ensembles making up the vector.  Must divide
+        evenly into dimensions
+    neurons_per_dimensions : int
+        Number of neurons in an ensemble will be this*subdimensions
+    vocab : Vocabulary, optional
+        The vocabulary to use to interpret this vector
+    direct : boolean
+        Whether or not to use direct mode for the neurons
```

Comment:
We don't need ```direct``` as a parameter, as it should be done by the config system (as with the other Networks)